langchain/src/experimental/masking/parser.ts

```diff
@@ -0,0 +1,139 @@
+import { MaskingTransformer } from "./transformer.js";
+import { MaskingParserConfig } from "./types.js";
+
+/**
+ * MaskingParser class for handling the masking and rehydrating of messages.
+ */
+export class MaskingParser {
+  private transformers: MaskingTransformer[];
+  private state: Map<string, string>;
+  private config: MaskingParserConfig;
+
+  constructor(config: MaskingParserConfig = {}) {
+    this.transformers = config.transformers || [];
```

Comment:
Ah ok I see you have it as an option here - also a bit more conventional for us to prefer `??` but doesn't really matter